backend/app/core/conf.py

```diff
@@ -1,19 +1,28 @@
 #!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 from functools import lru_cache
-from typing import Optional
 
-from pydantic import BaseSettings
+from pydantic import BaseSettings, root_validator
 
 
 class Settings(BaseSettings):
+    ENVIRONMENT: str
     # FastAPI
     TITLE: str = 'FastAPI'
-    VERSION: str = 'v0.0.1'
+    VERSION: str = 'v1'
     DESCRIPTION: str = "FastAPI Best Architecture"
-    DOCS_URL: Optional[str] = '/v1/docs'
-    REDOCS_URL: Optional[str] = None
-    OPENAPI_URL: Optional[str] = '/v1/openapi'
+    DOCS_URL: str = None
+    REDOCS_URL: str = None
+    OPENAPI_URL: str = None
+
+    @root_validator
+    def validator_api_url(cls, values):
+        # 开发环境和测试环境才显示文档
+        if values["ENVIRONMENT"] in ("development", "testing"):
```

Comment:
Hi, @downdawn 

I propose to keep the name simple or use Enum

E.g.:

```python 
if values["ENVIRONMENT"] in ("dev", "pro"):
```

Reply:
I think the full name would be more standardized
```
    DOCS_URL: str = '/v1/docs'
    REDOCS_URL: str = '/v1/redocs'
    OPENAPI_URL: str = '/v1/openapi'

    @root_validator
    def validator_api_url(cls, values):
        # 生产环境不开放API文档
        if values['ENVIRONMENT'] == 'production':
            values['OPENAPI_URL'] = None
        return values
```

What bothers me more is, what is the function of setting the v1 version? What if you want to upgrade some interfaces to v2?

It seems that it should not be written as fixed here
```
oauth2_schema = OAuth2PasswordBearer(tokenUrl='/v1/users/login')
```

Reply:
It is very common to use common abbreviations in development

version view: [fastapi](https://fastapi.tiangolo.com/advanced/wsgi/)

Set tokenUrl to a constant
```python
TOKEN_URL: str = '/v1/users/login'
```

Reply:
We should follow the FastAPI parameter types: [L50](https://github.com/tiangolo/fastapi/blob/c81e136d75f5ac4252df740b35551cf2afb4c7f1/fastapi/applications.py#L50)

```python
DOCS_URL: Optional[str] = '/v1/docs'
REDOCS_URL: Optional[str] = '/v1/redocs'
OPENAPI_URL: Optional[str] = '/v1/openapi'

@root_validator
def validator_api_url(cls, values):
    if values['ENVIRONMENT'] == 'pro':
        values['OPENAPI_URL'] = None
    return values

# Token Url
TOKEN_URL: str = '/v1/users/login'
```

Reply:
Agree

Reply:
close pr?

Reply:
Please update and submit the code